src/flow/multiroute_flow.jl

```diff
@@ -0,0 +1,182 @@
+  flow_algorithm::AbstractFlowAlgorithm     # keyword argument for algorithm
+  )
+  n = size(capacity_matrix,1)
+  return zero(T), zeros(T,n,n)
+end
+
+# Method for Kishimoto algorithm
+function multiroute_flow{T<:Number}(
+  flow_graph::DiGraph,                   # the input graph
+  source::Int,                           # the source vertex
+  target::Int,                           # the target vertex
+  capacity_matrix::AbstractArray{T,2},   # edge flow capacities
+  flow_algorithm::AbstractFlowAlgorithm, # keyword argument for algorithm
+  mrf_algorithm::KishimotoAlgorithm,     # keyword argument for algorithm
+  routes::Int                            # keyword argument for routes
+  )
+  return kishimoto(flow_graph, source, target, capacity_matrix, flow_algorithm, routes)
+end
+
+## Methods for Extended Multiroute Flow Algorithm
+#1 When the breaking points are not already known
+function multiroute_flow{T<:Number,R<:Real}(
+  flow_graph::DiGraph,                            # the input graph
+  source::Int,                                    # the source vertex
+  target::Int,                                    # the target vertex
+  capacity_matrix::AbstractArray{T,2},            # edge flow capacities
+  flow_algorithm::AbstractFlowAlgorithm,          # keyword argument for algorithm
+  mrf_algorithm::ExtendedMultirouteFlowAlgorithm, # keyword argument for algorithm
+  routes::R                                       # keyword argument for routes
+  )
+  return emrf(flow_graph, source, target, capacity_matrix, flow_algorithm, routes)
+end
+#2 When the breaking points are already known
+function multiroute_flow{T<:Number,R<:Real}(
+  breakingpoints::Vector{Tuple{T,T,Int}},         # vector of breaking points
+  routes::R,                                      # keyword argument for routes
+  flow_algorithm::AbstractFlowAlgorithm =         # keyword argument for algorithm
+      PushRelabelAlgorithm()
+  )
+  return intersection(breakingpoints,routes,flow_algorithm)
+end
+
+"""
+The generic multiroute_flow function will output two kinds of results:
+
+- When the number of routes is 0 or non-specified, the set of breaking points of
+the multiroute flow is returned.
+- Otherwise, a tuple with 1) the maximum flow and 2) the flow matrix. When the
+max-flow subroutine is the Boykov-Kolmogorov algorithm, the associated mincut is
+returned as a third output.
+
+When the input is a network, it requires the following arguments:
+
+- flow_graph::DiGraph                   # the input graph
+- source::Int                           # the source vertex
+- target::Int                           # the target vertex
+- capacity_matrix::AbstractArray{T,2}   # edge flow capacities
+- flow_algorithm::AbstractFlowAlgorithm # keyword argument for flow algorithm
+- mrf_algorithm::AbstractFlowAlgorithm  # keyword argument for multiroute flow algorithm
+- routes::R<:Number                     # keyword argument for the number of routes
```

Comment:
Forgot it in the doc part, sorry.

That being said, the elements' type in the capacity matrix is `T<:Number` (following the code in maximum_flow.jl). Since I need to convert this matrix into Float anyway, I am also moving the type T to `T<:Real` in multiroute_flow.jl